tests/ui_tools/test_boxes.py

```diff
@@ -251,6 +251,39 @@ def test_generic_autocomplete_emojis(self, write_box, text,
         typeahead_string = write_box.generic_autocomplete(text, state)
         assert typeahead_string == required_typeahead
 
+    @pytest.mark.parametrize(['typeaheads', 'suggestions', 'state',
+                              'expected_state', 'expected_typeahead',
+                              'is_truncated'], [
+      (['*zero*', '*one*', '*two*'], ['zero', 'one', 'two'], 1, 1,
+       '*one*', False),
+      (['*zero*', '*one*', '*two*'] * 4, ['zero', 'one', 'two'] * 4, 1,
+       1, '*one*', True),
+      (['*zero*', '*one*', '*two*'], ['zero', 'one', 'two'], None, None,
+       None, False),
+      (['*zero*', '*one*', '*two*'], ['zero', 'one', 'two'], 5, None, None,
+       False),
+      (['*zero*', '*one*', '*two*'], ['zero', 'one', 'two'], -5, None, None,
+       False),
+     ], ids=[
+        'less_than_10_typeaheads',
+        'more_than_10_typeaheads',
+        'invalid_state-None',
+        'invalid_state-greater_than_possible_index',
+        'invalid_state-less_than_possible_index',
+    ])
+    def test__process_typeaheads(self, write_box, typeaheads, suggestions,
+                                 state, expected_state, expected_typeahead,
+                                 is_truncated, mocker):
+        write_box.view.set_typeahead_footer = mocker.patch(
+                                'zulipterminal.ui.View.set_typeahead_footer')
+
+        typeahead = write_box._process_typeaheads(typeaheads, state,
+                                                  suggestions)
+
+        assert typeahead == expected_typeahead
+        write_box.view.set_typeahead_footer.assert_called_once_with(
+                            suggestions[:10], expected_state, is_truncated)
+
```

Comment:
I'm not sure what to do about the existing `generic_autocomplete` tests now, since now they span two different methods. Changing them to check for `_process_typeaheads` would be hard because the parameters would get complex. (Because we'd have to create lists of suggestions)